### PR TITLE
Make arguments override manifest, and more

### PR DESCRIPTION
**Pull request by rafie**
- Added ramp.sh to execute without installation,
- Added os_list metadata item.

### CODE REVIEW

File: RAMP/ramp.py [hunk 1]
```diff
@@ -84,17 +84,18 @@ def unpack(bundle):
 @click.option('--redis-pack-min-version', '-R', 'redis_pack_min_version', default=module_metadata.MIN_REDIS_PACK_VERSION, help='redis pack minimum version')
 @click.option('--config-command', '-cc', default=module_metadata.CONFIG_COMMAND, help='command used to configure module args at runtime')
 @click.option('--os', '-O', default=module_metadata.OS, help='build target OS (Darwin/Linux)')
+@click.option('--os-list', callback=comma_seperated_to_list, help='build target OS (Darwin/Linux)')
```
Comment (MeirShpilraien) [inline RAMP/ramp.py]: @rafie pls sync from master, we remove the os_list feature

Comment (MeirShpilraien) [reply RAMP/ramp.py]: Also there are default values like author which are not None, would those always be override?

File: RAMP/module_metadata.py [hunk 2]
```diff
@@ -23,10 +24,31 @@
 CONFIG_COMMAND = ""
 OVERIDE_COMMAND = []
 
-FIELDS = ["module_name", "module_file", "architecture", "version", "semantic_version",
-          "display_name", "author", "email", "description", "homepage", "license",
-          "command_line_args", "capabilities", "min_redis_version", "min_redis_pack_version",
-          "sha256", "commands", "os", "ramp_format_version", "config_command"]
+FIELDS = [
```
Comment (swilly22) [inline RAMP/module_metadata.py]: much clearer

File: RAMP/packer.py [hunk 3]
```diff
@@ -49,85 +50,97 @@ def archive(module_path, metadata, archive_name='module.zip'):
...
 
-    archive_file = zipfile.ZipFile(archive_name, 'w', zipfile.ZIP_DEFLATED)
     try:
-        archive_file.write(module_path, metadata["module_file"])
-        archive_file.write('module.json')
-        print(archive_name)
+        with zipfile.ZipFile(archive_name, 'w', zipfile.ZIP_DEFLATED) as archive_file:
+            archive_file.write(module_path, metadata["module_file"])
+            archive_file.write(jfile, arcname='module.json')
+            print(archive_name)
```
Comment (swilly22) [inline RAMP/packer.py]: Do we want this print?

File: RAMP/ramp.py [hunk 4]
```diff
@@ -68,34 +70,29 @@ def unpack(bundle):
...
-@click.option('--architecture', '-A', default=module_metadata.ARCHITECTURE, help='module compiled on i386/x86_64 arch')
-@click.option('--description', '-D', default=module_metadata.DESCRIPTION, help='short description')
-@click.option('--homepage', '-h', default=module_metadata.HOMEPAGE, help='module homepage')
-@click.option('--license', '-l', default=module_metadata.LICENSE, help='license')
-@click.option('--cmdargs', '-c', default=module_metadata.COMMAND_LINE_ARGS, help='module command line arguments')
-@click.option('--redis-min-version', '-r', 'redis_min_version', default=module_metadata.MIN_REDIS_VERSION, help='redis minimum version')
-@click.option('--redis-pack-min-version', '-R', 'redis_pack_min_version', default=module_metadata.MIN_REDIS_PACK_VERSION, help='redis pack minimum version')
-@click.option('--config-command', '-cc', default=module_metadata.CONFIG_COMMAND, help='command used to configure module args at runtime')
-@click.option('--os', '-O', default=module_metadata.OS, help='build target OS (Darwin/Linux)')
+@click.option('--display-name', '-d', default=None, help='name for display purposes')
```
Comment (swilly22) [inline RAMP/ramp.py]: Why did you removed all default values?

File: RAMP/packer.py (hunk 3 above)
Comment (rafie) [reply RAMP/packer.py]: Was there before. I think we use this output to find out what's the name of the archive is if we used expressions like {samantic-version} as part of the filename.

File: RAMP/ramp.py (hunk 4 above)
Comment (rafie) [reply RAMP/ramp.py]: First, there are two kind of arguments there: ones that correspond to RAMP fields and ones that don't. The latter kind (listed at the end) have default values. For the "field-args", we have to know whether they are passed by the user, as we want them to override default and manifest values. By giving them a default value of None we can detect that. Note that default values are provided by create_default_metadata(), and used to initialize the metadata structure before overriding it with manifest and command-line values.

**Pull request merged**